store/postgres/src/relational/ddl_tests.rs

```diff
@@ -60,6 +60,75 @@ fn check_eqv(left: &str, right: &str) {
     }
 }
 
+#[test]
+fn test_manual_index_creation_ddl() {
+    let layout = Arc::new(test_layout(BOOKS_GQL));
+
+    fn assert_generated_sql(
+        layout: Arc<Layout>,
+        entity_name: &str,
+        field_names: Vec<String>,
+        index_method: &str,
+        expected_format: &str,
+    ) {
+        let namespace = layout.site.namespace.clone();
+        let expected = expected_format.replace("{namespace}", namespace.as_str());
+
+        let (_, sql): (String, String) = generate_index_creation_sql(
+            layout.clone(),
+            entity_name,
+            field_names,
+            index::Method::from_str(index_method).unwrap(),
+            None,
+        )
+        .unwrap();
+
+        check_eqv(&expected, sql.trim());
+    }
+
+    const BTREE: &str = "btree"; // Assuming index::Method is the enum containing the BTree variant
+
+    assert_generated_sql(
+        layout.clone(),
+        "Book",
+        vec!["id".to_string()],
+        BTREE,
+        "create index concurrently if not exists manual_book_id on {namespace}.book using btree (\"id\")",
+    );
+
+    assert_generated_sql(
+        layout.clone(),
+        "Book",
+        vec!["content".to_string()],
+        BTREE,
+        "create index concurrently if not exists manual_book_content on {namespace}.book using btree (substring(\"content\", 1, 64))",
+    );
+
+    assert_generated_sql(
+        layout.clone(),
+        "Book",
+        vec!["title".to_string()],
+        BTREE,
+        "create index concurrently if not exists manual_book_title on {namespace}.book using btree (left(\"title\", 256))",
+    );
+
+    assert_generated_sql(
+        layout.clone(),
+        "Book",
+        vec!["page_count".to_string()],
+        BTREE,
+        "create index concurrently if not exists manual_book_page_count on {namespace}.book using btree (\"page_count\")",
+    );
+
+    assert_generated_sql(
+        layout.clone(),
+        "Book",
+        vec!["page_count".to_string(), "title".to_string()],
+        BTREE,
+        "create index concurrently if not exists manual_book_page_count_title on {namespace}.book using btree (\"page_count\", left(\"title\", 256))",
+    );
+}
```

Comment:
Very nice! It would be good to also add a test for an index on `Book.writtenBy` (should be GiST over `(written_by, block_range)`, a test for a explicitly including `block_range` like an index on `vec!["content", "block_range"]` and one for a partial index.